pallets/pallet-bank/src/tests.rs

```diff
@@ -398,3 +397,78 @@ fn incorrect_role_cannot_call_auditor_function() {
             assert!(Bank::check_total_issuance());
         });
 }
+
+#[test]
+fn auditor_cannot_unlock_invalid_id() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000)])
+        .build()
+        .execute_with(|| {
+            let charlie: AccountId = 3u32;
+            assert_ok!(Roles::register_role(&charlie, Role::Auditor));
+            assert_err!(
+                Bank::unlock_funds_auditor(RuntimeOrigin::signed(charlie), ALICE, 200),
+                crate::Error::<Runtime>::InvalidLockId
+            );
+        });
+}
+
+#[test]
+fn manager_can_set_interest_rate() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000)])
+        .build()
+        .execute_with(|| {
+            let charlie: AccountId = 3u32;
+            assert_ok!(Roles::register_role(&charlie, Role::Manager));
+            assert_ok!(Bank::set_interest_rate(RuntimeOrigin::signed(charlie), 500));
+            assert_eq!(InterestRate::<Runtime>::get(), Perbill::from_percent(5));
+        });
+}
+
+#[test]
+fn incorrect_role_cannot_set_interest_rate() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000)])
+        .build()
+        .execute_with(|| {
+            let charlie: AccountId = 3u32;
+            assert_ok!(Roles::register_role(&charlie, Role::Auditor));
+            assert_err!(
+                Bank::set_interest_rate(RuntimeOrigin::signed(charlie), 500),
+                pallet_roles::Error::<Runtime>::IncorrectRole
+            );
+            assert_err!(
+                Bank::set_interest_rate(RuntimeOrigin::signed(ALICE), 500),
+                pallet_roles::Error::<Runtime>::IncorrectRole
+            );
+            assert!(InterestRate::<Runtime>::get().is_zero());
+        });
+}
+
+#[test]
+fn pay_interest() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000_000_000)])
+        .build()
+        .execute_with(|| {
+            let charlie: AccountId = 3u32;
+            assert_ok!(Roles::register_role(&charlie, Role::Manager));
+            assert_ok!(Bank::set_interest_rate(RuntimeOrigin::signed(charlie), 500));
+            assert_ok!(Bank::stake_funds(
```

Comment:
Can create a helper function to stake funds in the test.rs 
All tests can use this function to stake
i.e. 
fn stake(user: AccountId, amount: Balance) {
// get user's reserve balance
// call the stake function with the user and amount
// cal on-finalize to ensure the fund is unlocked
// check users reserve balance is incremented
}